### PR TITLE
Fix: support writes to zero

### PR DESCRIPTION
**Pull request by odesenfans**
Problem: we encounter an "Edge bottom not found in preimage" error when running some blocks. This problem was diagnosed as writes to zero where the last node of the storage proof was an edge node. A write to zero to a specific address will clear the path to that leaf node, so the storage proof for this key will always be incomplete. It turns out that the OS will still look for the bottom node in the preimage to verify the node hash.

Solution: if we encounter such a case, we now fetch an additional key that is designed to follow the edge path. This way, we get the bottom node in the preimage and the OS can compute the hash it wants to compute.

Issue Number: N/A

## Type

- [ ] feature
- [x] bugfix
- [ ] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Breaking changes?

- [ ] yes
- [x] no

### CODE REVIEW

File: crates/bin/prove_block/src/rpc_utils.rs [hunk 1]
```diff
@@ -150,40 +249,35 @@ pub(crate) async fn get_storage_proofs(
...
-fn merge_chunked_storage_proofs(proofs: Vec<PathfinderProof>) -> PathfinderProof {
+/// Returns a modified key that follows the specified edge path.
+/// This function is used to work around an issue where the OS fails if it encounters a
+/// write to 0 and the last node in the storage proof is an edge node of length 1.
+/// In this situation the OS will still look up the node in the preimage and will fail
+/// on a "Edge bottom not found in preimage" error.
+/// To resolve this, we fetch the storage proof for a node that follows this edge in order
+/// to get the bottom node in the preimage and resolve the issue.
+fn get_key_following_edge(key: Felt, height: Height, edge_path: &EdgePath) -> Felt {
+    let mask = edge_path.value.to_bigint() << (251 - height.0);
```
Comment (notlesh) [inline crates/bin/prove_block/src/rpc_utils.rs:274]: Could use a constant instead of `251`

Comment (notlesh) [reply crates/bin/prove_block/src/rpc_utils.rs:274]: Maybe also assert that `height.0 < 251`?

**Review by notlesh — APPROVED** [review]
Looks good other than a couple of cosmetic suggestions.

Another question, though: is there any equivalent of this for the class tree?

**Comment by odesenfans** [comment]
> Looks good other than a couple of cosmetic suggestions.
> 
> Another question, though: is there any equivalent of this for the class tree?

There is no equivalent yet, I don't think this is necessary. There is no way to delete a class, so there is no "write to 0" for the contract and class tries. Furthermore, you are always guaranteed to always go 0 -> something or X -> Y, so there will always be a valid proof for at least the current state.

**Pull request merged**